src/pages/index.js

```diff
@@ -129,28 +110,100 @@ function closeModal(modal) {
   modal.classList.remove("modal_opened");
   document.removeEventListener('keydown', handleEscKey);
   modal.removeEventListener('mousedown', handleOverlayClick);
+
+  const formEl = modal.querySelector("form");
+  if (formEl) {
+    resetForms(formEl);
+    resetValidation(formEl, settings);
+  }
 };
 
-function resetCardForm() {
-  cardFormElement.reset();
-  cardModalSubmitBtn.disabled = true;
-  cardModalSubmitBtn.classList.add("modal__submit-btn_inactive");
+function handleEscKey(evt) {
+  if (evt.key === 'Escape') {
+    const activeModal = document.querySelector('.modal_opened');
+    if (activeModal) {
+      closeModal(activeModal, settings);
+    }
+  }
+}
+
+function handleOverlayClick(evt) {
+  if (evt.target.classList.contains('modal_opened')) {
+    closeModal(evt.target, evt.forms);
+  }
 }
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api.editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value})
+  .then((userInfo) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+  })
+  .then(() => closeModal(editModal, editFormElement))
+  .catch(console.error)
+  .finally(() => {
+    setButtonText(submitBtn, false);
+  });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {name: cardNameInput.value, link: cardLinkInput.value}
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  closeModal(cardModal);
-  resetCardForm(cardModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api.addNewCard({name: cardNameInput.value, link: cardLinkInput.value})
+  .then(renderCard)
+  .then(() => closeModal(cardModal, cardFormElement))
+  .catch(console.error)
+  .finally(() => {
+    setButtonText(submitBtn, false);
+  });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api.editUserAvatar({ avatar: avatarLinkInput.value })
+  .then((userPic) => {
+    profileImage.src = userPic.avatar;
+    closeModal(avatarModal, avatarFormElement);
+  })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**

Reply:
Updated.